chapters/annotations.tex

```diff
@@ -1272,6 +1272,66 @@ \section{Graphical User Interface}\label{annotations-for-the-graphical-user-inte
 \end{lstlisting}
 \end{example}
 
+A connector component declaration may have the following annotation:
+\begin{lstlisting}[language=modelica]
+annotation(mustBeConnected = "message");
+\end{lstlisting}%
+\annotationindex{mustBeConnected}
+
+If non-empty it indicates that it is an error if the connector is not connected from the outside (for a conditional connector this check is only active if the connector is enabled).
+For an array of connectors it applies separately to each element.
+
+\begin{nonnormative}
+Note that this annotation is only intended for non-causal connectors, see \cref{restrictions-of-connections-and-connectors}.
+It is in particular suited for stream-connectors, \cref{stream-connectors}.
+\end{nonnormative}
+
+\begin{example}
+This can be used for some optional connectors that should be connected when conditionally enabled.
+\begin{lstlisting}[language=modelica]
+partial model PartialWithSupport
+  Flange_a flange_a;
+  Flange_b flange_b;
+  Support support($\ldots$)
+    if useSupport annotation(mustBeConnected = "Support connector should be connected if activated.");
+  $\ldots$
+end PartialWithSupport;
+\end{lstlisting}
+\end{example}
+
+A connector component declaration may have the following annotation:
+\begin{lstlisting}[language=modelica]
+annotation(mayOnlyConnectOnce = "message");
+\end{lstlisting}%
+\annotationindex{mayOnlyConnectOnce}
+
+If non-empty it indicates that it is an error if it is connected from the outside and:
+\begin{itemize}
+\item For non-stream connectors the connection set has more than two elements.
+\item For stream connectors, \cref{stream-connectors}, the connection set has more than two elements whose flow variable may be negative.
```

Comment:
What does this restriction really mean? Is it a condition to be checked during simulation?

Reply:
During translation using min-values as in https://specification.modelica.org/master/derivation-of-stream-equations.html#connection-of-3-stream-connectors-where-one-mass-flow-rate-is-identical-to-zero-n-3-and

Reply:
Outdated thread replaced by new comment. Resolving.